### PR TITLE
[Bug][Api] Complete Data workflow remove duplicates.

### LINKED ISSUE #10849: [Bug][Api] Complete Data workflow remove duplicates.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When complete data, the income date format is `2022-06-01 00:02:10, 2022-06-01 00:02:10, 2022-06-02 00:02:10`, and there are duplicate scheduling dates.

### What you expected to happen

1.When complementing, scheduling does not allow duplicates.

### How to reproduce

1.run workflow domplete data.
2.enter data format is `2022-06-01 00:02:10, 2022-06-01 00:02:10, 2022-06-02 00:02:10`

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hstdream**
## Purpose of the pull request

1.When complementing, scheduling does not allow duplicates

fix #10849 

## Brief change log
1.remove string date duplicates
## Verify this pull request
Run complete data workflow, enter date format is `2022-06-01 00:02:10, 2022-06-01 00:02:10, 2022-06-02 00:02:10`

### CODE REVIEW

**Comment by hstdream** [comment]
#10849

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java [hunk 1]
```diff
@@ -988,8 +988,7 @@ private String removeDuplicates(String scheduleTimeList) {
         HashSet<String> removeDate = new HashSet<String>();
         List<String> resultList = new ArrayList<String>();
         if (StringUtils.isNotEmpty(scheduleTimeList)) {
-            String[] dateArrays = scheduleTimeList.split(COMMA);
-            List<String> dateList = Arrays.asList(dateArrays);
+            List<String> dateList  = Arrays.stream(scheduleTimeList.split(COMMA)).map(String::trim).collect(Collectors.toList());
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: I think we should distinct dataList.

Comment (hstdream) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: ok,changed.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java [hunk 2]
```diff
@@ -984,15 +984,12 @@ private boolean isValidateScheduleTime(String schedule) {
...
+    private static String removeDuplicates(String scheduleTimeList) {
         HashSet<String> removeDate = new HashSet<String>();
-        List<String> resultList = new ArrayList<String>();
         if (StringUtils.isNotEmpty(scheduleTimeList)) {
-            String[] dateArrays = scheduleTimeList.split(COMMA);
-            List<String> dateList = Arrays.asList(dateArrays);
-            removeDate.addAll(dateList);
-            resultList.addAll(removeDate);
-            return String.join(COMMA, resultList);
+            List<String> inputDateList  = Arrays.stream(scheduleTimeList.split(COMMA)).map(String::trim).collect(Collectors.toList());
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: I understand that the problem you fixed is to delete the space before and after, not the repetition. Am I right?

Comment (hstdream) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: no, here is the problem of repeated scheduling time.

Comment (hstdream) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: Space will cause the problem of repeated scheduling time.

Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]:
```suggestion
            Set<String> dateSet  = Arrays.stream(scheduleTimeList.split(COMMA)).map(String::trim).collect(Collectors.toSet());
```

Comment (caishunfeng) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: And the name `removeDate` seems ambiguous, suggestion: dateSet.

Comment (hstdream) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java]: ok

**Review by kezhenxu94 — APPROVED** [review]
I think this is a code refactor, not a bug fix?

**Comment by hstdream** [comment]
> I think this is a code refactor, not a bug fix?

Input date is code refactor.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
So this PR will use array list to replace the old hash set logic, then the output will contains duplicates schedule time.

It's better to change the method name, since this method just trim the schedule time, there is no duplicate logic.

**Comment by ruanwenjun** [comment]
So this PR will use array list to replace the old hash set logic, then the output will contains duplicates schedule time.
 
 It's better to change the method name, since this method just trim the schedule time, there is no deduplicate logic.

**Comment by hstdream** [comment]
> > So this PR will use array list to replace the old hash set logic, then the output will contains duplicates schedule time.
> > It's better to change the method name, since this method just trim the schedule time, there is no deduplicate logic.


There will be no duplicate.

**Comment by ruanwenjun** [comment]
> > > So this PR will use array list to replace the old hash set logic, then the output will contains duplicates schedule time.
> > > It's better to change the method name, since this method just trim the schedule time, there is no deduplicate logic.
> 
>  There will be no duplicate.

But you don't use hash set here, is the GitHub bug?
image

**Comment by hstdream** [comment]
> > So this PR will use array list to replace the old hash set logic, then the output will contains duplicates schedule time.
> > It's better to change the method name, since this method just trim the schedule time, there is no deduplicate logic.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Comment by ruanwenjun** [comment]
Please don't cherry-pick this PR, this PR need to be fix again.

**Comment by ruanwenjun** [comment]
Will be fixed by #10868

**Pull request merged**